### PR TITLE
Drop support for v0 and v1 blocks

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
Drops support for v0 and v1 blocks and consolidates all code into v2.

**TODO**
- [x] Investigate Performance regression indicated by benchmarks
- [x] Figure out what to do with buffers

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by joe-elliott** [comment]
This is ready to go, but leaving draft status up b/c I want to merge #806 first. There will be some small conflicts.

File: tempodb/encoding/v2/data_reader.go [hunk 1]
```diff
@@ -23,41 +27,97 @@ var constDataHeader = &dataHeader{}
...
+		if err != nil {
+			return nil, nil, err
+		}
 	}
 
 	return pagesBuffer, buffer, nil
 }
 
 func (r *dataReader) Close() {
-	r.dataReader.Close()
```
Comment (zalegrala) [inline tempodb/encoding/v2/data_reader.go:60]: Do we still need to do something here to close this out?

Comment (joe-elliott) [reply tempodb/encoding/v2/data_reader.go:60]: Nope, b/c there is no longer an encapsulated data reader.

**Review by zalegrala — APPROVED** [review]

File: tempodb/encoding/v2/data_reader.go [hunk 2]
```diff
@@ -23,41 +27,97 @@ var constDataHeader = &dataHeader{}
...
+		}
+
+		compressedPages = append(compressedPages, page.data)
+	}
+
+	// prepare pagesBuffer
+	if cap(pagesBuffer) < len(compressedPages) {
+		// extend pagesBuffer
+		diff := len(compressedPages) - cap(pagesBuffer)
+		pagesBuffer = append(pagesBuffer[:cap(pagesBuffer)], make([][]byte, diff)...)
```
Comment (annanay25) [inline tempodb/encoding/v2/data_reader.go:96]: `append()` will copy over all the items from old to new array right? and the resultant pagesBuffer array might have a cap larger than `len(compressedPages)`, worth just creating a new array with `cap(pagesBuffer) = len(compressedPages)` ? The reuse is primarily useful when `cap(pagesBuffer) > len(compressedPages)` I guess

**Review by annanay25 — APPROVED** [review]
LGTM, just a minor non-blocking nit

File: tempodb/encoding/v2/data_reader.go (hunk 2 above)
Comment (joe-elliott) [reply tempodb/encoding/v2/data_reader.go:96]: > and the resultant pagesBuffer array might have a cap larger than len(compressedPages)
I believe after this line len(pagesBuffer) = cap(pagesBuffer) = len(compressedPages)

> worth just creating a new array with cap(pagesBuffer) = len(compressedPages)
by using whatever currently exists in pagesBuffer we reduce byte slice allocations. after this line the first N entries in pagesBuffer already have allocated byte slices that can be reused in `ReadAllWIthBuffer`. if we recreated the slice we would lose that.

**Pull request merged**